src/search3_cleanup.erl

```diff
@@ -0,0 +1,50 @@
+-module(search3_cleanup).
+
+-export([
+    clear_unreachable_indexes/1
+]).
+
+-include("search3.hrl").
+-include_lib("couch/include/couch_db.hrl").
+-include_lib("fabric/include/fabric2.hrl").
+
+clear_unreachable_indexes(Db) ->
+    Batch = config:get_integer("search3", "index_delete_batch", 300),
+    #{
+        db_prefix := DbPrefix
+    } = Db,
+    Remaining = fabric2_fdb:transactional(Db, fun(TxDb) ->
+        #{
+            tx := Tx
+        } = TxDb,
+        Indexes = search3_util:list_indexes(TxDb),
+        ExistingSigs  = [Index#index.sig || Index <- Indexes],
+        SigsFromList = get_sigs_from_list(Tx, DbPrefix),
+        ClearList = lists:filter(fun (Sig) ->
+            not lists:member(Sig, ExistingSigs)
+        end, SigsFromList),
+        ListToClear = lists:sublist(ClearList, Batch),
```

Comment:
we're only clearing every 300 indexes in case we reach a timeout. It's arbitrary and can be configurable.

Reply:
I'd reduce that so we can be more confident each loop clears something.